packages/transaction-controller/src/TransactionController.ts

```diff
@@ -3786,6 +3794,20 @@ export class TransactionController extends BaseController<
     this.#updateTransactionInternal(
       { transactionId, skipHistory: true },
       (txMeta) => {
+        const userFeeLevel = txMeta.userFeeLevel as GasFeeEstimateLevel;
+
+        if (Object.values(GasFeeEstimateLevel).includes(userFeeLevel)) {
+          txMeta.txParams.maxFeePerGas =
+            (gasFeeEstimates as FeeMarketGasFeeEstimates)?.[userFeeLevel]
+              ?.maxFeePerGas ||
+            (gasFeeEstimates as LegacyGasFeeEstimates)?.[userFeeLevel] ||
+            (gasFeeEstimates as GasPriceGasFeeEstimates)?.gasPrice;
+
+          txMeta.txParams.maxPriorityFeePerGas = (
```

Comment:
Should we only be setting this if the transaction type is fee market or if 1559 is supported?

We wouldn't want to set this on a legacy transaction for example, even if the poller returned a value.